### PR TITLE
mbl-cloud-client: use new script configure scheme

### PR DESCRIPTION
**Pull request by jonathanhaigh-arm**
In the near future we're going to remove our bootflags partition. That
probably means that the path to the bootflags will change ("/mnt/flags"
doesn't seem appropriate if "/mnt/flags" isn't actually a mount point).

In preparation for that, mbl-cloud-client's update scripts have been
updated to take the path to the boot flags, and a few other things as
configure options. Update the recipe to pass appropriate values for
these options.

We can also now remove arm_update_local_config.sh because it was only
there as a hack to work around not having a proper configure mechanism
for the update scripts.

Now that the scripts are "configured" we have to get them from the build
directory rather than the source directory so update do_install() for
that.

**Jenkins Links**
* http://jenkins.mbed-linux.arm.com/view/jh/job/jh-test/127/ (failed on imx8mm - looks like a temporary failure to access a repo).
* http://jenkins.mbed-linux.arm.com/job/jh-test/128/ (succeeded).
* http://jenkins.mbed-linux.arm.com/view/jh/job/jh-test/131/ (in progress; after changes for review).

**Lava Links**
* http://e120856-lin/results.php?image_version=custom&image_number=131&custom_version=jh-test&lava_query=&lava_owner=&failures=false&submitter=jonhai01&sort=by_job
  * Some of these tests failed due to timeouts writing to flash, and some of them succeeded after re-submission. I don't see any failures that are due to the changes here, except possibly increasing the probability of timeouts due to making the images a bit bigger.

**Related PRs**
* https://github.com/ARMmbed/mbl-core/pull/200

**Notes**
* Need to update mbl-core SRCREV before merging this PR.

### CODE REVIEW

File: meta-mbl-apps/recipes-connectivity/mbl-cloud-client/mbl-cloud-client.bb [hunk 1]
```diff
@@ -123,14 +128,14 @@ do_compile() {
 
 do_install() {
     install -d "${D}/opt/arm"
-    install "${S}/cloud-services/mbl-cloud-client/__${TARGET}/${RELEASE_TYPE}/mbl-cloud-client" "${D}/opt/arm"
-    install "${S}/cloud-services/mbl-cloud-client/__${TARGET}/${RELEASE_TYPE}/pelion-provisioning-util" "${D}/opt/arm"
+    output_dir="${S}/cloud-services/mbl-cloud-client/__${TARGET}"
+    install "${output_dir}/${RELEASE_TYPE}/mbl-cloud-client" "${D}/opt/arm"
```
Comment (diego-sueiro) [inline meta-mbl-apps/recipes-connectivity/mbl-cloud-client/mbl-cloud-client.bb]: Can you please enforce the file permissions on the installation? I suppose you want `755`.

File: meta-mbl-apps/recipes-connectivity/mbl-cloud-client/mbl-cloud-client.bb [hunk 2]
```diff
@@ -123,14 +128,14 @@ do_compile() {
...
+    install "${output_dir}/${RELEASE_TYPE}/pelion-provisioning-util" "${D}/opt/arm"
```
Comment (diego-sueiro) [inline meta-mbl-apps/recipes-connectivity/mbl-cloud-client/mbl-cloud-client.bb]: ditto.

File: meta-mbl-apps/recipes-connectivity/mbl-cloud-client/mbl-cloud-client.bb (hunk 1 above)
Comment (jonathanhaigh-arm) [reply meta-mbl-apps/recipes-connectivity/mbl-cloud-client/mbl-cloud-client.bb]: 7d62cd637a7bd35c3903a9b8f32cd9c41e0b880d

File: meta-mbl-apps/recipes-connectivity/mbl-cloud-client/mbl-cloud-client.bb (hunk 2 above)
Comment (jonathanhaigh-arm) [reply meta-mbl-apps/recipes-connectivity/mbl-cloud-client/mbl-cloud-client.bb]: 7d62cd637a7bd35c3903a9b8f32cd9c41e0b880d

**Pull request merged**